### PR TITLE
Interaction - Missions: Chains of Promathia Chapter 1

### PR DESCRIPTION
**Pull request by claywar**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

* Adds additional default dialogue changes and switches to Mission variables for Rites of Life (Reasoning behind using vars instead of missionStatus is to reserve space for once we decipher its impact on the mission log
* Audits, updates, and converts Below the Arks and The Mothercrystals to Interaction Framework.
* Implements the sealing of memories for Incomplete areas
* Implement memory sealing from left apparatus.
* Adds CoP helpers lua for reusing above implementation with both Below the Arks and Mothercrystals 
* Adds distance check to Right-side large apparatus (<= 2.71 yalms)

### CODE REVIEW

File: scripts/globals/bcnm.lua [hunk 1]
```diff
@@ -649,11 +649,11 @@ local function checkReqs(player, npc, bfid, registrant)
...
-        [ 800] = function() return ( (cop == mi.cop.BELOW_THE_ARKS and copStat==1) or (cop == mi.cop.THE_MOTHERCRYSTALS and not player:hasKeyItem(xi.ki.LIGHT_OF_DEM))     ) end, -- PM1-3: The Mothercrystals
+        [ 800] = function() return ( (cop == mi.cop.BELOW_THE_ARKS) or (cop == mi.cop.THE_MOTHERCRYSTALS and not player:hasKeyItem(xi.ki.LIGHT_OF_DEM))     ) end, -- PM1-3: The Mothercrystals
         [ 801] = function() return ( player:hasKeyItem(xi.ki.CENSER_OF_ANTIPATHY)                                                                                          ) end, -- ENM: You Are What You Eat
-        [ 832] = function() return ( (cop == mi.cop.BELOW_THE_ARKS and copStat==1) or (cop == mi.cop.THE_MOTHERCRYSTALS and not player:hasKeyItem(xi.ki.LIGHT_OF_MEA))     ) end, -- PM1-3: The Mothercrystals
+        [ 832] = function() return ( (cop == mi.cop.BELOW_THE_ARKS) or (cop == mi.cop.THE_MOTHERCRYSTALS and not player:hasKeyItem(xi.ki.LIGHT_OF_MEA))     ) end, -- PM1-3: The Mothercrystals
```
Comment (zach2good) [inline scripts/globals/bcnm.lua]: Can we keep the formatting here?

File: scripts/zones/Hall_of_Transference/npcs/_0e6.lua [hunk 2]
```diff
@@ -22,7 +22,7 @@ entity.onTrigger = function(player, npc)
     if player:getCharVar("DemChipRegistration") == 1 then
         player:messageSpecial(ID.text.NO_RESPONSE_OFFSET + 6) -- Device seems to be functioning correctly.
     else
-        player:startEvent(167) -- Hexagonal Cones
+        player:startEvent(167) -- Hexagonal Cones -- FREAKING EVENT 165
```
Comment (zach2good) [inline scripts/zones/Hall_of_Transference/npcs/_0e6.lua]: ?

File: scripts/zones/Hall_of_Transference/npcs/_0e8.lua [hunk 3]
```diff
@@ -22,7 +22,7 @@ entity.onTrigger = function(player, npc)
     if player:getCharVar("MeaChipRegistration") == 1 then
         player:messageSpecial(ID.text.NO_RESPONSE_OFFSET + 6) -- Device seems to be functioning correctly.
     else
-        player:startEvent(163) -- Hexagonal Cones
+        player:startEvent(163) -- Hexagonal Cones -- 167 here
```
Comment (zach2good) [inline scripts/zones/Hall_of_Transference/npcs/_0e8.lua]: ?

**Review by zach2good — APPROVED** [review]

File: scripts/missions/cop/1_3_The_Mothercrystals.lua [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+                    mission:setVar(player, 'Option', xi.cop.helpers.promyvionCrags.MEA)
+                end,
+            },
+        },
+
+        [xi.zone.SPIRE_OF_MEA] =
+        {
+            onEventFinish =
+            {
+                [32001] = xi.cop.helpers.spireEventFinish,
```
Comment (zach2good) [inline scripts/missions/cop/1_3_The_Mothercrystals.lua]: All very neat and tidy 👍

**Comment by claywar** [comment]
I'm about as happy as I can be with this at the moment, appreciate feedback.

File: scripts/missions/cop/1_3_The_Mothercrystals.lua [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+    return xi.cop.helpers.largeApparatusOnTrigger(mission, player, npc)
+end
+
+local largeApparatusOnEventFinish = function(player, csid, option, npc)
+    xi.cop.helpers.largeApparatusOnEventFinish(mission, player, csid, option, npc)
+end
+
+local spireEventFinish = function(player, csid, option, npc)
+    xi.cop.helpers.spireEventFinish(mission, player, csid, option, npc)
+end
```
Comment (zach2good) [inline scripts/missions/cop/1_3_The_Mothercrystals.lua:42]: Good compromise :+1:

**Review by zach2good — APPROVED** [review]

**Pull request merged**